stats/opentelemetry/opentelemetry.go

```diff
@@ -0,0 +1,239 @@
+	// used only for generic methods, and not registered methods.
+	MethodAttributeFilter func(string) bool
+}
+
+// DialOption returns a dial option which enables OpenTelemetry instrumentation
+// code for a grpc.ClientConn.
+//
+// Client applications interested in instrumenting their grpc.ClientConn should
+// pass the dial option returned from this function as a dial option to
+// grpc.Dial().
+//
+// For the metrics supported by this instrumentation code, a user needs to
+// specify the client metrics to record in metrics options. A user also needs to
+// provide an implementation of a MeterProvider. If the passed in Meter Provider
+// does not have the view configured for an individual metric turned on, the API
+// call in this component will create a default view for that metric.
+func DialOption(mo MetricsOptions) grpc.DialOption {
+	csh := &clientStatsHandler{mo: mo}
+	csh.initializeMetrics()
+	return joinDialOptions(grpc.WithChainUnaryInterceptor(csh.unaryInterceptor), grpc.WithStreamInterceptor(csh.streamInterceptor), grpc.WithStatsHandler(csh))
+}
+
+// ServerOption returns a server option which enables OpenTelemetry
+// instrumentation code for a grpc.Server.
+//
+// Server applications interested in instrumenting their grpc.Server should pass
+// the server option returned from this function as an argument to
+// grpc.NewServer().
+//
+// For the metrics supported by this instrumentation code, a user needs to
+// specify the client metrics to record in metrics options. A user also needs to
+// provide an implementation of a MeterProvider. If the passed in Meter Provider
+// does not have the view configured for an individual metric turned on, the API
+// call in this component will create a default view for that metric.
+func ServerOption(mo MetricsOptions) grpc.ServerOption {
+	ssh := &serverStatsHandler{mo: mo}
+	ssh.initializeMetrics()
+	return grpc.StatsHandler(ssh)
+}
+
+// callInfo is information pertaining to the lifespan of the RPC client side.
+type callInfo struct {
+	target string
+
+	method string
+}
+
+type callInfoKey struct{}
+
+func setCallInfo(ctx context.Context, ci *callInfo) context.Context {
+	return context.WithValue(ctx, callInfoKey{}, ci)
+}
+
+// getCallInfo returns the callInfo stored in the context, or nil
+// if there isn't one.
+func getCallInfo(ctx context.Context) *callInfo {
+	ci, _ := ctx.Value(callInfoKey{}).(*callInfo)
+	return ci
+}
+
+// rpcInfo is RPC information scoped to the RPC attempt life span client side,
+// and the RPC life span server side.
+type rpcInfo struct {
+	mi *metricsInfo
+}
+
+type rpcInfoKey struct{}
+
+func setRPCInfo(ctx context.Context, ri *rpcInfo) context.Context {
+	return context.WithValue(ctx, rpcInfoKey{}, ri)
+}
+
+// getRPCInfo returns the rpcInfo stored in the context, or nil
+// if there isn't one.
+func getRPCInfo(ctx context.Context) *rpcInfo {
+	ri, _ := ctx.Value(rpcInfoKey{}).(*rpcInfo)
+	return ri
+}
+
+func removeLeadingSlash(mn string) string {
```

Comment:
Optional: consider removing and inlining the code directly.

Reply:
Prefer this, as it's used 5 times. But I guess it's strings.TrimLeft vs. removeLeadingSlash. Both convey similar semantic meanings. Let me know if you feel strongly about this, and I'll change it, I'm pretty indifferent.

Reply:
Looking more at the usages, I'm a bit concerned about how it's needed throughout the code.  Is it possible to centralize where it's used to where things enter the system so that we aren't calling it from many different places, like the top of `TagRPC`?

Reply:
Hmmm good point. I persist a method string on client and server side, and call removeLeadingSlash on it every time it's read. I'll just persist the stripped method instead of the full method. I'll still leave this helper though, but I can remove it too I'm indifferent but I like what the method name says haha.